api/cirrus_ci_service.proto

```diff
@@ -264,6 +264,49 @@ service CirrusCIService {
   }
   rpc GenerateCacheDownloadURLs (CacheKey) returns (GenerateURLsResponse) {
   }
+
+  // Multipart cache upload
+  rpc MultipartCacheUploadCreate(MultipartCacheUploadCreateRequest) returns (MultipartCacheUploadCreateResponse);
+  rpc MultipartCacheUploadPart(MultipartCacheUploadPartRequest) returns (MultipartCacheUploadPartResponse);
+  rpc MultipartCacheUploadCommit(MultipartCacheUploadCommitRequest) returns (MultipartCacheUploadCommitResponse);
+}
+
+message MultipartCacheUploadCreateRequest {
+  CacheKey cache_key = 1;
+}
+
+message MultipartCacheUploadCreateResponse {
+  string upload_id = 1;
+}
+
+message MultipartCacheUploadPartRequest {
+  CacheKey cache_key = 1;
+  string upload_id = 2;
+  uint32 part_number = 3;
+}
+
+message MultipartCacheUploadPartResponse {
```

Comment:
Can it be `GenerateURLResponse`?

Reply:
`GenerateURLResponse` does not account for multiple values in a single header, which pre-signing S3 client does.

Also, this this seems to go against the [Create Unique Protos per Method](https://protobuf.dev/programming-guides/api/#unique-protos) best practice.

Reply:
I'm just asking because `GenerateURLResponse` contains a pre-signed URL and we don't encounter multiple values per header situation.

Reply:
See https://github.com/cirruslabs/cirrus-cli/pull/774/commits/a013f13f3e6103db2565abddf1d3b5082bd4bc4d.